pkg/autoscaler/stats_scraper.go

```diff
@@ -117,41 +122,54 @@ func newServiceScraperWithClient(
 
 // Scrape calls the destination service then sends it
 // to the given stats channel.
-func (s *ServiceScraper) Scrape() (*StatMessage, error) {
+func (s *ServiceScraper) Scrape(statsCh chan<- *StatMessage) {
 	readyPodsCount, err := resources.FetchReadyAddressCount(s.endpointsLister, s.namespace, s.scrapeTargetService)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to get endpoints")
+		s.logger.Errorf("Failed to get endpoints: %v", err)
+		return
 	}
 
 	if readyPodsCount == 0 {
-		return nil, nil
+		s.logger.Debug("Do not scrape stats because no ready pods.")
+		return
 	}
 
-	stat, err := s.sClient.Scrape(s.url)
-	if err != nil {
-		return nil, err
+	sampleSize := populationMeanSampleSize(readyPodsCount)
+	if sampleSize < 1 {
+		s.logger.Debug("Do not scrape stats because sample size is less than 1.")
+		return
 	}
 
-	// Assumptions:
-	// 1. Traffic is routed to pods evenly.
-	// 2. A particular pod can stand for other pods, i.e. other pods have
-	//    similar concurrency and QPS.
-	//
-	// Hide the actual pods behind scraper and send only one stat for all the
-	// customer pods per scraping. The pod name is set to a unique value, i.e.
-	// scraperPodName so in autoscaler all stats are either from activator or
-	// scraper.
-	extrapolatedStat := Stat{
-		Time:                             stat.Time,
-		PodName:                          scraperPodName,
-		AverageConcurrentRequests:        stat.AverageConcurrentRequests * float64(readyPodsCount),
-		AverageProxiedConcurrentRequests: stat.AverageProxiedConcurrentRequests * float64(readyPodsCount),
-		RequestCount:                     stat.RequestCount * int32(readyPodsCount),
-		ProxiedRequestCount:              stat.ProxiedRequestCount * int32(readyPodsCount),
+	for i := 0; i < sampleSize; i++ {
+		go func() {
```

Comment:
Shouldn't we wait for these goroutines to finish? Otherwise you'll have the "Scrape-called-multiple-times-if-latency-increases" problem.

Reply:
What object do you want to make it wait for these goruntines to finish? The `collector`?